hudi-spark-datasource/hudi-spark-common/src/main/java/org/apache/hudi/internal/DataSourceInternalWriterHelper.java

```diff
@@ -31,15 +31,17 @@
 import org.apache.hudi.exception.HoodieException;
 import org.apache.hudi.table.HoodieSparkTable;
 import org.apache.hudi.table.HoodieTable;
+
+import avro.shaded.com.google.common.annotations.VisibleForTesting;
```

Comment:
would we use `com.google.common.annotations.VisibleForTesting` to align the codebase?

Reply:
+1

Reply:
guess it gave me illigal imports checkstyle error. I wanted to understand the guidelines in general on how to go about VisibleForTesting things since this annotation is part of illegal imports. 

To test some private methods, how one should approach
- If we can use VisibleFoTesting annotation, its much simpler. But guess its not allowed. So, I can think of a hacky solution. Create a test class inheriting from the class of interest. make the method of interest protected and expose/overide from tests. this way you can get access to this method of interest. But as you could see, this is not a neat solution.

Reply:
>To test some private methods, how one should approach

May be subjective, but I have good reason to believe, private methods should not be tested directly

Reply:
may be I can do some work around for this. I might have to read the extra metadata from commit metadata and validate the extra metadata. but I feel many times, this might lead to spending more time or complex test code to validate something very simple.